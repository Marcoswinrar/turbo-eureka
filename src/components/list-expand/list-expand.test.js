import { fireEvent, render, screen } from "@testing-library/react"
import { act } from "react-dom/test-utils"
import ListExpand from "."

describe("Componente <ListExpand/>", () => {
  it("Deve renderizar component de Lista Expansivel sem problemas", () => {
    const { container } = render(<ListExpand />)

    expect(container).toMatchSnapshot()
    expect(container.firstChild).not.toHaveClass("open")
  })

  it("Deve expandir ao clicar encima do header", () => {
    render(<ListExpand />)
    const header = screen.queryByText(/Hooks - useEffect/)

    act(() => {
      fireEvent.click(header)
    })

    expect(screen.queryByText(/Estudo sobre hooks - useEffect/))
  })
})