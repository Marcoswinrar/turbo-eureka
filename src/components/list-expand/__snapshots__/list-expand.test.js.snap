// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente <ListExpand/> Deve renderizar component de Lista Expansivel sem problemas 1`] = `
<div>
  <div
    class="sc-hKMtZM HhcTB"
  >
    <section
      class="sc-eCYdqJ kYIXbH"
    >
      <header
        class="sc-gKXOVf jwCYRe"
      >
        <h3
          class="sc-jSMfEi bWEXfk"
        >
          Hooks - useEffect
        </h3>
      </header>
    </section>
  </div>
</div>
`;
